### PR TITLE
Ensure messages from bg threads are displayed

### LINKED ISSUE #10864: Output is not updated from background thread
From microsoft/vscode#150924

Run this cell:

```ts
import threading, time

def network_call():
    for i in range(20):
        print(i)
        time.sleep(1)
    
threading.Thread(target=network_call).start()
```

The `0` is printed, but no updates come after that

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #10864

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

File: src/kernels/execution/cellExecutionQueue.ts [hunk 1]
```diff
@@ -120,10 +123,18 @@ export class CellExecutionQueue implements Disposable {
         const kernelConnection = await this.session;
         this.queueOfCellsToExecute.forEach((exec) => traceCellMessage(exec.cell, 'Ready to execute'));
         while (this.queueOfCellsToExecute.length) {
+            // Dispose the previous cell execution.
+            // We should keep the last cell execution alive, as we could get messages even after
+            // cell execution has completed, leaving it alive, allows it to process the messages.
+            // If we run another cell, then the background messages from previous cell will be picked up
+            // by the next cell execution (which is now the last cell execution).
+            this.lastCellExecution?.dispose();
```
Comment (sadasant) [inline src/kernels/execution/cellExecutionQueue.ts:131]: Love this approach!

**Review by sadasant — APPROVED** [review]
Clean design, and you even fixed the tests ❤️👏

File: src/kernels/execution/cellExecution.ts [hunk 2]
```diff
@@ -171,7 +171,6 @@ export class CellExecution implements IDisposable {
         // Begin the request that will modify our cell.
         this.execute(this.codeOverride || this.cell.document.getText().replace(/\r\n/g, '\n'), session)
             .catch((e) => this.completedWithErrors(e))
-            .finally(() => this.dispose())
             .catch(noop);
```
Comment (amunger) [inline src/kernels/execution/cellExecution.ts:174]: already handled exceptions, so this catch can be removed?

Comment (DonJayamanne) [reply src/kernels/execution/cellExecution.ts:174]: nope, this is to ensure we don't have any floating promises.
else nodejs complains about these promises.
eg things could fail in finally, hence we need a catch after that...

**Comment by GF-Huang** [comment]
@DonJayamanne  Is it really merged?

image

image

- Jupyter v2022.6.1201981810

**Comment by rchiodo** [comment]
Merged goes into our main branch. Main branch is currently on 2022.7

**Pull request merged**